packages/rum_sdk/lib/src/transport/rum_transport.dart

```diff
@@ -6,28 +6,41 @@ import 'package:rum_sdk/src/transport/rum_base_transport.dart';
 import 'package:rum_sdk/src/transport/task_buffer.dart';
 import 'package:rum_sdk/src/models/payload.dart';
 
-
 class RUMTransport extends BaseTransport {
   final String collectorUrl;
   final String apiKey;
+  final String? sessionId;
   TaskBuffer<dynamic>? _taskBuffer;
-  RUMTransport({required this.collectorUrl, required this.apiKey, maxBufferLimit = 30 }) {
+
+  RUMTransport({
+    required this.collectorUrl,
+    required this.apiKey,
+    this.sessionId,
+    maxBufferLimit = 30,
+  }) {
     _taskBuffer = TaskBuffer(maxBufferLimit);
   }
 
   @override
   Future<void> send(Payload payload) async {
-    final headers = {'Content-Type': 'application/json', 'x-api-key': apiKey};
-    final response = await _taskBuffer?.add((){
-       return http.post(
-         Uri.parse(collectorUrl),
-         headers: headers,
-         body: jsonEncode(payload.toJson()),
-       );
+    final sessionId = this.sessionId;
+
+    final headers = {
+      'Content-Type': 'application/json',
+      'x-api-key': apiKey,
+      if (sessionId != null) 'x-faro-session-id': sessionId,
```

Comment:
I had to add this. Otherwise I got this error from Faro in Grafana Cloud:
`missing X-Faro-Session-Id header, activate the SessionInstrumentation`